src/angularjs/src/app/places/compare/compare.controller.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @ngdoc controller
+ * @name pfb.analysis-jobs.detail.controller:CompareController
+ *
+ * @description
+ * Controller for showing details about an analysis job
+ *
+ */
+(function() {
+    'use strict';
+
+    /** @ngInject */
+    function CompareController($stateParams, Neighborhood, AnalysisJob, $log, $q, $state) {
+        var ctl = this;
+
+        initialize();
+
+        function initialize() {
+            ctl.places = new Array(3);
+            ctl.getPlace = getPlace;
+            ctl.clearSelection = clearSelection;
+
+            getPlaces([$stateParams.place1, $stateParams.place2, $stateParams.place3]);
+        }
+
+        function getPlaces(uuids) {
+            var promises = _.map(uuids, function(uuid, offset) {
+                return getPlace(offset, uuid);
+            });
+
+            // do not display any place until all places have been retrieved
+            $q.all(promises).then(function(results) {
+                ctl.places = results;
```

Comment:
Delaying display in any case would require knowing when all promises have either returned or failed, which as far as I can see would still require a promises-aware refactor, which is why I went with this.